examples/OSMPDummyFunction/OSMPDummyFunction.cpp

```diff
@@ -0,0 +1,878 @@
+     * \param roll Global roll reference of rotation.
+     * \param rx Local x-coordinate after rotation. 
+     * \param ry Local y-coordinate after rotation.
+     * \param rz Local z-coordinate after rotation.
+     */ 
+void rotatePoint3D(double x, double y, double z,double yaw,double pitch,double roll,double &rx,double &ry,double &rz) // TODO: Find out why 3D rotation doesn't work. Using 2D rotation instead.
+{
+    double matrix[3][3];
+    double cos_yaw = cos(yaw*M_PI/180);
+    double cos_pitch = cos(pitch*M_PI/180);
+    double cos_roll = cos(roll*M_PI/180);
+    double sin_yaw = sin(yaw*M_PI/180);
+    double sin_pitch = sin(pitch*M_PI/180);
+    double sin_roll = sin(roll*M_PI/180);
+
+    matrix[0][0] = cos_yaw*cos_pitch;  matrix[0][1]=cos_yaw*sin_pitch*sin_roll - sin_yaw*cos_roll; matrix[0][2]=cos_yaw*sin_pitch*cos_roll + sin_yaw*sin_roll;
+    matrix[1][0] = sin_yaw*cos_pitch;  matrix[1][1]=sin_yaw*sin_pitch*sin_roll + cos_yaw*cos_roll; matrix[1][2]=sin_yaw*sin_pitch*cos_roll - cos_yaw*sin_roll;
+    matrix[2][0] = -sin_pitch;         matrix[2][1]=cos_pitch*sin_roll;                            matrix[2][2]=cos_pitch*cos_roll;
+
+    rx = matrix[0][0] * x + matrix[0][1] * y + matrix[0][2] * z;
+    ry = matrix[1][0] * x + matrix[1][1] * y + matrix[1][2] * z;
+    rz = matrix[2][0] * x + matrix[2][1] * y + matrix[2][2] * z;
+
+}
+
+/*!
+     * \brief 2D rotation matrix function. Calculates local X,Y coordinates and relative yaw angle from reference.  
+     * \param x Global relative x-coordinate.
+     * \param y Global relative y-coordinate.
+     * \param yaw Global yaw reference of rotation.
+     * \param rx Local x-coordinate after rotation. 
+     * \param ry Local y-coordinate after rotation.
+     * \param ryaw Local yaw angle to reference after rotation.
+     */
+void rotatePoint2D(double x, double y, double yaw, double &rx, double &ry, double &ryaw)
+{
+    double cos_yaw = cos(yaw * M_PI/180);
+    double sin_yaw = sin(yaw * M_PI/180);
+
+    rx = cos_yaw*x + sin_yaw*y;
+    ry = -sin_yaw*x + cos_yaw*y; 
+
+    ryaw = (atan2(ry,rx) * 180/M_PI);
+}
+
+void COSMPDummyFunction::evaluateAEB(double &rel_x, double &rel_y, double &rel_z, double &rel_vx, double &rel_vy, double &rel_vz, double &rel_yaw, double &decelRequest)
+{
+    // Return if relative is positive. (AKA not approaching target)
+    if (rel_vx >= 0){
+        return;
+    }
+
+    // Function variables. Thinking of adding these to class/input instead ?? 
+
+    double pbTTClimit = 2.5; // prebrake 
+    double pbRequest = -3.0;
+
+    double fbTTClimit = 1.5; // fullbrake
+    double fbRequest = -5.0; 
+
+    // TODO add information to Sensordata so that this could be derived
+    double collisionWidth = 1.5; // ~Width/2 of target (Could be derived with getting dimension of target) 
+    double distanceOffset = 4.5; // ~Length/2 of target + ~length/2 of ego. (Since Sensorview calculates distance from base of ego to base of target) 

```

Comment:
All these as class members I think is a good idea.

Reply:
Ok. What is the best way to initialize them?

Reply:
Directly in the header feels like the simplest option but is there a better way ?

Reply:
Nothing wrong in doing it like that in my opinion